kotlin-generator/src/helpers.ts

```diff
@@ -0,0 +1,240 @@
+import {
+    ArrayType,
+    Base64PrimitiveType,
+    BigIntPrimitiveType,
+    BoolPrimitiveType,
+    BytesPrimitiveType,
+    CnpjPrimitiveType,
+    CpfPrimitiveType,
+    DatePrimitiveType,
+    DateTimePrimitiveType,
+    EmailPrimitiveType,
+    EnumType,
+    FloatPrimitiveType,
+    HexPrimitiveType,
+    IntPrimitiveType,
+    JsonPrimitiveType,
+    MoneyPrimitiveType,
+    OptionalType,
+    StringPrimitiveType,
+    StructType,
+    Type,
+    TypeReference,
+    UIntPrimitiveType,
+    UrlPrimitiveType,
+    UuidPrimitiveType,
+    VoidPrimitiveType,
+    XmlPrimitiveType,
+} from "@sdkgen/parser";
+
+export function generateEnum(type: EnumType) {
+    return `enum class ${type.name} { ${type.values.map(x => mangle(x.value)).join(", ")} }\n`;
+}
+
+export function generateClass(type: StructType) {
+    return `data class ${type.name}(\n${type.fields
+        .map(field => {
+            let fieldDesc = "";
+            if (field.type.constructor === DatePrimitiveType) {
+                fieldDesc += "        @JsonAdapter(DateAdapter::class)\n";
+            } else if (field.type.constructor === DateTimePrimitiveType) {
+                fieldDesc += "        @JsonAdapter(DateTimeAdapter::class)\n";
+            }
+            fieldDesc += `        val ${mangle(field.name)}: ${generateKotlinTypeName(field.type)}`;
+            return fieldDesc;
+        })
+        .join(",\n")}\n    )\n`;
+}
+
+export function generateErrorClass(error: string) {
+    return `class ${error}(message: String) : Error(message)\n`;
+}
+
+export function generateKotlinTypeName(type: Type): string {
+    switch (type.constructor) {
+        case IntPrimitiveType:
+        case UIntPrimitiveType:
+        case MoneyPrimitiveType:
+            return "Int";
+
+        case FloatPrimitiveType:
+            return "Double";
+
+        case BigIntPrimitiveType:
+            return "BigInt";
+
+        case DatePrimitiveType:
+        case DateTimePrimitiveType:
+            return "Calendar";
+
+        case BoolPrimitiveType:
+            return "Boolean";
+
+        case BytesPrimitiveType:
+            return "ByteArray";
+
+        case StringPrimitiveType:
+        case CpfPrimitiveType:
+        case CnpjPrimitiveType:
+        case EmailPrimitiveType:
+        case UrlPrimitiveType:
+        case UuidPrimitiveType:
+        case HexPrimitiveType:
+        case Base64PrimitiveType:
+        case XmlPrimitiveType:
+            return "String";
+
+        case VoidPrimitiveType:
+            return "Unit";
+
+        case JsonPrimitiveType:
+            return "JsonObject";
```

Comment:
This should be `JsonValue`, as the `json` type can be array or a string, for example, only just objects

Reply:
Gson has: [JsonArray](https://www.javadoc.io/static/com.google.code.gson/gson/2.8.6/com.google.gson/com/google/gson/JsonArray.html), [JsonElement](https://www.javadoc.io/static/com.google.code.gson/gson/2.8.6/com.google.gson/com/google/gson/JsonElement.html), [JsonNull](https://www.javadoc.io/static/com.google.code.gson/gson/2.8.6/com.google.gson/com/google/gson/JsonNull.html), [JsonObject](https://www.javadoc.io/static/com.google.code.gson/gson/2.8.6/com.google.gson/com/google/gson/JsonObject.html) or [JsonPrimitive](https://www.javadoc.io/static/com.google.code.gson/gson/2.8.6/com.google.gson/com/google/gson/JsonPrimitive.html).

JsonObject is the most flexible, as it extends JsonElement.

Reply:
So it should be a JsonElement